js/mercado.js

```diff
@@ -168,14 +193,21 @@ module.exports = class mercado extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        if (type === 'market')
-            throw new ExchangeError (this.id + ' allows limit orders only');
-        let method = 'privatePostPlace' + this.capitalize (side) + 'Order';
         let order = {
             'coin_pair': this.marketId (symbol),
-            'quantity': amount,
-            'limit_price': price,
         };
+        let method = 'privatePostPlace' + this.capitalize (side) + 'Order';
+        if (type === 'limit') {
+            order['limit_price'] = price;
+            order['quantity'] = amount;
+        } else {
+            method = 'privatePostPlaceMarket' + this.capitalize (side) + 'Order';
+            if (side === 'buy') {
+                order['cost'] = (amount * price).toFixed (5);
```

Comment:
This is a logical error, unfortunately, because market orders don't require a price in general. So, we need to reimplement this with a warning, like shown here: https://github.com/ccxt/ccxt/blob/master/js/cex.js#L325

Reply:
You haven't completely fixed it, unfortunately, because the `price` should be required only when submitting the `market buy`. It is not required for a `market sell`. So we need to place that warning lower a bit.

Reply:
Can't see `cost` anywhere in their docs...